### PR TITLE
alfred,batctl: don't override PKG_CONFIG 

### LINKED ISSUE #237690: Build failure: batctl for riscv64
### Steps To Reproduce
Failed to cross compile batctl package for riscv64 using nixpkgs on **Ubuntu**.  Nix is not able to meat the dependency for the package:

`Makefile:116: *** No libnl-3.0 development libraries found!.  Stop.`

While the package successfully cross compiles on a **NixOS** machine. 

1. Install nix environment on a Ubuntu machine. and cross compile batctl package for riscv64. Below is the command I am using:
`   nix-build '<nixpkgs>' -A pkgsCross.riscv64.batctl`

### Build log

attached

### Metadata

nix-info not availble on Ubuntu

[log.txt.txt](https://github.com/NixOS/nixpkgs/files/11742539/log.txt.txt)

### PR DESCRIPTION
**Pull request by MaxHearnden**
###### Description of changes

fixes https://github.com/NixOS/nixpkgs/issues/237690

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)

there was nothing to be built
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by MaxHearnden** [comment]
@fpletz can you review this, ofborg did not pick up the change

File: pkgs/os-specific/linux/batman-adv/alfred.nix [hunk 1]
```diff
@@ -15,7 +15,7 @@ stdenv.mkDerivation rec {
   buildInputs = [ gpsd libcap libnl ];
 
   preBuild = ''
-    makeFlags="PREFIX=$out PKG_CONFIG=${pkg-config}/bin/${pkg-config.targetPrefix}pkg-config"
+    makeFlags="PREFIX=$out PKG_CONFIG=${buildPackages.pkg-config}/bin/${buildPackages.pkg-config.targetPrefix}pkg-config"
```
Comment (Artturin) [inline pkgs/os-specific/linux/batman-adv/alfred.nix:18]:
```suggestion
    makeFlags="PREFIX=$out PKG_CONFIG=$PKG_CONFIG"
```
apply to both

Comment (MaxHearnden) [reply pkgs/os-specific/linux/batman-adv/alfred.nix]: I instead let the makefile inherit it directly from the environment

**Review by fpletz — APPROVED** [review]
Thanks!

**Pull request merged**